src/SessionMonitor/index.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { userapiPath } from '../localconf';
+import getReduxStore from '../reduxStore';
+import { logoutAPI } from '../actions';
+
+class SessionMonitor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { mostRecentActivityTimestamp: Date.now() };
+  }
+
+  componentDidMount() {
+    window.addEventListener('mousedown', this.updateUserActivity, false);
+    window.addEventListener('keypress', this.updateUserActivity, false);
+    setTimeout(this.refreshSession, this.props.refreshSessionTime); // check session every X min
+  }
+
+  updateUserActivity = () => {
+    this.setState({ mostRecentActivityTimestamp: Date.now() });
+  }
+
+  refreshSession = () => {
+    // If user has been inactive for Y min
+    if (Date.now() - this.state.mostRecentActivityTimestamp > this.props.inactiveTimeLimit) {
+      getReduxStore().then((store) => {
+        store.dispatch(logoutAPI());
+      });
+    } else {
+      fetch(userapiPath); // hitting Fence endpoint refreshes token
+      setTimeout(this.refreshSession, this.props.refreshSessionTime);
+    }
+  }
+
```

Comment:
Ok - calling `setTimeout` multiple times accomplishes the same thing as `setInterval` if you want to do it that way - I expected `sessionMontor.refreshSession()` to be this thing that just does a `fetch` rather than part of a state machine, but either way works :-p